src/main/java/uk/gov/hmcts/reform/preapi/controllers/MediaServiceController.java

```diff
@@ -89,6 +96,24 @@ public ResponseEntity<LiveEventDTO> getLiveEvents(@PathVariable String liveEvent
         return ResponseEntity.ok(data);
     }
 
+    @GetMapping("/vod")
+    @PreAuthorize("hasAnyRole('ROLE_SUPER_USER', 'ROLE_LEVEL_1', 'ROLE_LEVEL_2', 'ROLE_LEVEL_3', 'ROLE_LEVEL_4')")
+    public ResponseEntity<PlaybackDTO> getVod(
+        @RequestParam UUID recordingId,
+        @RequestParam(required = false) String mediaService
+    ) throws InterruptedException {
+        // check recording exists + authed
+        recordingService.findById(recordingId);
+
+        var service = mediaServiceBroker.getEnabledMediaService(mediaService);
+
+        // TODO dont rely on naming convention, link asset name in db
```

Comment:
Not for this PR but maybe we need to consider some kind of naming utility class to centralise these naming conventions (same for shortened/sanitised UUIDs)

Reply:
Yeah definitely agree with this